search/search-client-connect/src/main/java/io/camunda/search/connect/os/OpensearchConnector.java

```diff
@@ -45,24 +48,44 @@ public final class OpensearchConnector {
 
   private final ConnectConfiguration configuration;
   private final ObjectMapper objectMapper;
+  private final PluginRepository pluginRepository;
+
+  private AwsCredentialsProvider credentialsProvider;
 
   public OpensearchConnector(final ConnectConfiguration configuration) {
-    this(configuration, new JacksonConfiguration(configuration).createObjectMapper());
+    this(
+        configuration,
+        new JacksonConfiguration(configuration).createObjectMapper(),
+        DefaultCredentialsProvider.create(),
+        new PluginRepository());
   }
 
   public OpensearchConnector(
-      final ConnectConfiguration configuration, final ObjectMapper objectMapper) {
+      final ConnectConfiguration configuration,
+      final ObjectMapper objectMapper,
+      final AwsCredentialsProvider credentialsProvider,
```

Comment:
❓ Why was this change necessary?

Reply:
It was very hard to test/mock: because method `boolean shouldCreateAWSBasedTransport()` explicitly invokes `DefaultCredentialsProvider.create()`. See, the  `DefaultCredentialsProvider` initializes in chain `EnvironmentVariableCredentialsProvider`, `SystemPropertyCredentialsProvider`, and few others. This means, that people (like myself), who have a set of configs for AWS accounts as env vars, will have AWS initialized, thus making impossible to test local instance of OpenSearch.

So I decided that since there are multiple implementations of the `AwsCredentialsProvider`, it might not hurt to move it into constructor. Please keep in mind, that the backwards compatibility is respected, and when not defined, setting `DefaultCredentialsProvider.create()`.

WDYT?